libraries/provider_firewall_iptables.rb

```diff
@@ -17,123 +17,109 @@
 
-        # prints all the firewall rules
-        # pp new_resource.subresources
-        log_current_iptables
-        if active?
-          Chef::Log.info("#{new_resource} already enabled.")
-        else
-          Chef::Log.debug("#{new_resource} is about to be enabled")
-          shell_out!('iptables -P INPUT DROP')
-          shell_out!('iptables -P OUTPUT DROP')
-          shell_out!('iptables -P FORWARD DROP')
-
-          shell_out!('ip6tables -P INPUT DROP')
-          shell_out!('ip6tables -P OUTPUT DROP')
-          shell_out!('ip6tables -P FORWARD DROP')
-          Chef::Log.info("#{new_resource} enabled.")
-          new_resource.updated_by_last_action(true)
+        %w(iptables ip6tables).each do |svc|
+
+          # must create empty file for service to start
+          file "create empty /etc/sysconfig/#{svc}" do
+            path "/etc/sysconfig/#{svc}"
+            content '# created by chef to allow service to start'
+            not_if { ::File.exists?("/etc/sysconfig/#{svc}") }
+          end
+
+          service svc do
+            action [:enable, :start]
+          end
         end
       end
     end
 
-    def action_disable
-      if active?
-        shell_out!('iptables -P INPUT ACCEPT')
-        shell_out!('iptables -P OUTPUT ACCEPT')
-        shell_out!('iptables -P FORWARD ACCEPT')
-        shell_out!('iptables -F')
-
-        shell_out!('ip6tables -P INPUT ACCEPT')
-        shell_out!('ip6tables -P OUTPUT ACCEPT')
-        shell_out!('ip6tables -P FORWARD ACCEPT')
-        shell_out!('ip6tables -F')
-        Chef::Log.info("#{new_resource} disabled")
-        new_resource.updated_by_last_action(true)
-      else
-        Chef::Log.debug("#{new_resource} already disabled.")
-      end
+    action :restart do
+      next if disabled?(new_resource)
+
+      # prints all the firewall rules
+      log_iptables
+
+      # ensure it's initialized
+      new_resource.rules Hash.new unless new_resource.rules
+      ensure_default_rules_exist(new_resource.rules)
+
+      %w(iptables ip6tables).each do |iptables_type|
+        iptables_filename = "/etc/sysconfig/#{iptables_type}"
+        # ensure a file resource exists with the current iptables rules
+        begin
+          iptables_file = run_context.resource_collection.find(file: iptables_filename)
+        rescue
+          iptables_file = file iptables_filename do
+            action :nothing
+          end
+        end
+        iptables_content = new_resource.rules[iptables_type].sort_by { |k,v| v }.map { |k,v| k }.join("\n")
+        iptables_file.content "#{iptables_content}\n"
+        iptables_file.run_action(:create)
+
+        # if the file was changed, restart iptables
+        if iptables_file.updated_by_last_action?
+          service_affected = service iptables_type do
+            action :nothing
+          end
 
-      service service_name do
-        action [:disable, :stop]
+          new_resource.notifies(:restart, service_affected, :delayed)
```

Comment:
would a :reload not be better for updating your iptables config?
a restart would do a stop and start of iptables, potentialy resetting the established/related sessions. and give a (minor) window in which there is no firewall.
where as a reload would apply the new rules without the above mentioned reset of established/related sessions, and without the short window of allowing all traffic. 
i think this option is only available with iptables, though it might depends on the OS vesion.